apps/arweave/src/ar_chain_stats.erl

```diff
@@ -0,0 +1,173 @@
+-module(ar_chain_stats).
+
+-behaviour(gen_server).
+
+-include_lib("arweave/include/ar.hrl").
+-include_lib("arweave/include/ar_chain_stats.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-export([log_fork/2, get_forks/1]).
+
+-export([start_link/0, init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+%%%===================================================================
+%%% Public interface.
+%%%===================================================================
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+log_fork([], _ForkRootB) ->
+    %% No fork to log
+    ok;
+log_fork(Orphans, ForkRootB) ->
+    gen_server:cast(?MODULE, {log_fork, Orphans, ForkRootB, os:system_time(millisecond)}).
+
+%% @doc Returns all forks that have been logged since the given start time
+%% (system time in seconds)
+get_forks(StartTime) ->
+    case catch gen_server:call(?MODULE, {get_forks, StartTime}) of
+		{'EXIT', {timeout, {gen_server, call, _}}} ->
+			{error, timeout};
+		Reply ->
+			Reply
+	end.
+
+%%%===================================================================
+%%% Generic server callbacks.
+%%%===================================================================
+init([]) ->
+    process_flag(trap_exit, true),
+	ok = ar_kv:open(filename:join(?ROCKS_DB_DIR, "forks_db"), forks_db),
+    {ok, #{}}.
+
+handle_call({get_forks, StartTime}, _From, State) ->
+    {ok, ForksMap} = ar_kv:get_range(forks_db, integer_to_binary(StartTime * 1000)),
+    %% Sort forks by their key (the timestamp when they were detected)
+    SortedForks = lists:sort(maps:to_list(ForksMap)),
+    Forks = [binary_to_term(Fork) || {_Timestamp, Fork} <- SortedForks],
+    {reply, Forks, State};
+handle_call(_Request, _From, State) ->
+    {reply, ok, State}.
+
+handle_cast({log_fork, Orphans, ForkRootB, ForkTime}, State) ->
+    log_fork(Orphans, ForkRootB, ForkTime),
+    {noreply, State};
+handle_cast(_Msg, State) ->
+    {noreply, State}.
+
+handle_info(_Info, State) ->
+    {noreply, State}.
+
+terminate(_Reason, _state) ->
+    ok.
+
+%%%===================================================================
+%%% Private functions.
+%%%===================================================================
+log_fork(Orphans, ForkRootB, ForkTime) ->
+    Fork = create_fork(Orphans, ForkRootB, ForkTime),
+    ar_kv:put(forks_db, integer_to_binary(ForkTime), term_to_binary(Fork)),
```

Comment:
integer_to_binary(2) > integer_to_binary(11); I think, we need << ForkTime:64 >> here

Reply:
Yikes good catch. I added a test and fixex: https://github.com/ArweaveTeam/arweave/pull/581/commits/0542b4e1254d1d50b751c3b1589fb995e9b7f3a8